src/tasks/gitea/gitea-drone-oauth.ts

```diff
@@ -0,0 +1,212 @@
+    try {
+      const response = (await client.readNamespacedSecret(this.k8s_secret_name, this.k8s_namespace)).body
+      return response.data as DroneSecret
+    } catch (e) {
+      return undefined
+    }
+  }
+  async authorizeOAuth() {
+    const options: AxiosRequestConfig = {
+      params: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        client_id: this.oauthData.clientId,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        redirect_uri: `${this.droneUrl}/login`,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        response_type: 'code',
+      },
+      maxRedirects: 1,
+      auth: {
+        username: env.GITEA_USER,
+        password: env.GITEA_PASSWORD,
+      },
+    }
+
+    console.log('Authorizing OAuth application')
+
+    let authorizeResponse
+    try {
+      authorizeResponse = await axios.get(`${this.giteaUrl}/login/oauth/authorize`, options)
+    } catch (error) {
+      console.log('Authorization already granted or something went wrong')
+      console.error(error.config)
+      console.error('Options used in Axios request', options)
+      return
+    }
+
+    const authorizeHeaderCookies: string[] = authorizeResponse.headers['set-cookie']
+
+    // Loop over cookies and find the _csrf cookie and retrieve the value
+    this.csrfToken = authorizeHeaderCookies
+      .filter((cookie) => {
+        return cookie.includes(this.csrfCookieName) // Find cookie that contains _csrf
+      })[0]
+      .split(';') // Split that cookie into multiple parts, as a single cookie can contain multiple key/values ';' separated
+      .map((c) => c.trim())
+      .filter((cookie) => {
+        return cookie.includes(this.csrfCookieName) // Find the key/value pair with _csrf as key
+      })
+      .map((cookie) => {
+        return cookie.substring(this.csrfCookieName.length + 1) // Retrieve value for '_csrf'-key
+      })[0]
+
+    console.log('Granting authorization')
+
+    const grantOptions: AxiosRequestConfig = {
+      method: 'POST',
+      url: `${this.giteaUrl}/login/oauth/grant`,
+      headers: {
+        cookie: authorizeHeaderCookies
+          .map((cookie) => {
+            return cookie.split(';')[0]
+          })
+          .join('; '),
+        Cookie: authorizeHeaderCookies
+          .map((cookie) => {
+            return cookie.split(';')[0]
+          })
+          .join('; '),
+      },
+      maxRedirects: 1,
+      auth: {
+        username: env.GITEA_USER,
+        password: env.GITEA_PASSWORD,
+      },
+      // Data for this post query must be stringified https://github.com/axios/axios#using-applicationx-www-form-urlencoded-format
+      data: querystring.stringify({
+        _csrf: this.csrfToken,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        client_id: `${this.oauthData.clientId}`,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        redirect_uri: `${this.droneUrl}/login`,
```

Comment:
${this.droneUrl}/login is used in many places

can we define variable droneLoginUrl ?